nibabel/freesurfer/mghformat.py

```diff
@@ -459,9 +459,13 @@ class MGHImage(SpatialImage):
     """ Class for MGH format image
     """
     header_class = MGHHeader
-    files_types = (('image', '.mgh'),)
+    files_types = (('image', '.mgh'),
+                   ('image', '.mgz'))
```

Comment:
That looks to me like how it's working. I feel like a conversation was had on this subject a while back, but I'm guessing it got lost from the PR after a rebase. I believe this allowed us to avoid a decorator or new class variable just to accommodate `.mgz`, but perhaps there is a fairly easy solution:

Add a default `SpatialImage` class variable `alternate_extensions = ()`, update it in `MGHImage` (possibly through the `ImageOpener.register_ext_from_image` decorator) and rewrite `SpatialImage.is_valid_extensions`:

    @classmethod
    def is_valid_extension(klass, ext):
        valid_exts = tuple(ft[1] for ft in klass.files_types) + klass.alternate_extensions
        return ext.lower() in valid_exts